assignments/this.js

```diff
@@ -1,26 +1,74 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. Implicit Binding (this is used when you invoke a function and it is telling the this function to look on the left side of the dot. So if I have an object that's const me and it has an attribute called name, and I invoke a function that goes me.sayName, then the me on the left side of the dot is what the This statement will use. If I don't do this I get undefined )
+* 2. Explicit Binding (Allows you to directly point to/say what you want the This function to grab from. So instead of looking to the left of the dot, you write your const and do .call(insert thing you want your This to pull from) You can also use apply instead of cost if you're passing in an array, and you can use bind like apply/call but instead of immediately invoking the previous function it will return a new function to invoke later).
+* 3. New Binding (this is used to create whole new objects based off a similar object we already made. So we create an object with a function like const animal = function (color, name, type) and this using the this function in the attributes. You can create new objects quick and easy, (very DRY) but creating new objects and typying in the New function. so Const dog = new Animal (x, y, y) and JS creates it behind the scenes.)
```

Comment:
not necessarily based off an object we already made, but you are binding/assigning the this keyword with a new object made from a constructor function (template)

Reply:
when you are done with the mvp why don't you go back and try the different explicit binding types